docs/content/addons/custom-helm-config.md

```diff
@@ -0,0 +1,67 @@
++++
+title = "Custom Helm Chart Configuration"
+icon = "fa-solid fa-eye"
++++
+
+A default helm chart configuration is provided with a ConfigMap `default-helm-addons-config`. This ConfigMap contains helm chart URL and version for each addon.
+If a user want to customize the default configuration with different helm chart repository URL or version for one or more addons, the user can create a ConfigMap with their customized addon configuration and provide it as a reference in the `Cluster` object.
+
+The helm chart configuration for an addon included in the custom ConfigMap will be installed on the cluster.
+If configuration for an addon is not included in the the customized addon configuration ConfigMap, it will be defauled from the `default-helm-addons-config` configmap.
+
+The content of the ConfigMap must follow following specification.
+
+```yaml
+  addon-name: |
+    ChartName: addon-chart-name
+    ChartVersion: 1.0.0
+    RepositoryURL: https://my-chart-repository.example.com/charts/
+```
+
+Example configmap `custom-helm-addons-config.yaml`:
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: <CLUSTER_NAME>-helm-addons-config
+  namespace: <CLUSTER_NAMESPACE>
+  labels:
+    clusterctl.cluster.x-k8s.io/move: ""
```

Comment:
not sure if this label is strictly needed. According to the [Move documentation](https://cluster-api.sigs.k8s.io/developer/providers/contracts/clusterctl.html?#move), object with either clusterctl.cluster.x-k8s.io/move: "" or direct/indirect `ownerReference` will be moved. 
I am not sure if ownerReference is not explicitly added, it will be added indirectly when the configmap is referenced by the `Cluster` object.

Reply:
The ownerRef won't be added automatically, but I think the controller in CAREN should add it.

This is similar to what CAREN does with [CCM/CSI Secrets](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/53e00c521d2d7474ccb94a1fcd9cec761c6dd47c/pkg/handlers/generic/lifecycle/csi/nutanix/handler.go#L111) and adds an ownerRef.
Then when all clusters that own are deleted the Secret (CM in this case) would be garbageCollected.

Reply:
Agree. 
is CAREN mutating webhook is the right place to the owner reference?